docs/misc/ConfigDB_state.md

```diff
@@ -0,0 +1,89 @@
+
+### From Unknown
+
+On auth_data.AuthenticationDataManager.connect(), we check the actual DB mode change internal state as follows:
+
+* Unknown-> NotConfigured   use "no access control" mode for vmdk-opsd operations
+* Unknown-> SingleNode      open the DB and keep going (backward compat, local DB)
+* Unknown-> MultiNode        open the DB and keep going (shared DB via symlink)
+* Unknown-> BrokenLink      ERROR - throw exception
+
+
+### From NotConfigured
+
+Manual transition from NotConfigured is done by the admin CLI.
+Automatic transition from NotConfigured is done by Discovery (see below)
+
+* NotConfigured -> NotConfigured    Keep going.
+* NotConfigured -> SingleNode       Forced by by creating a new DB in /etc/vmware/vmdkops (AuthenticationDataManager.new_db(). When discovered, we just connect to the DB, log an INFO message and keep going.
+* NotConfigured -> MultiNode         Forced by creating the DB on a shared datastore and creating symlink in /etc/vmware/vmdkops. When discovered, we just connect to the DB, log an INFO message and keep going. (same as in NotConfigured->SingleNode)
+* NotConfigured -> BrokenLink       ERROR - throw exception
+
+### BrokenLink handling
+BrokenLink is an error condition. We do not know what was the admin intent so we fail all Docker volume requests until the situation is resolved by the admin using `config rm --local` command
+
+* BrokenLink -> NotConfigured - by admin `config rm` command
+* BrokenLink -> BrokenLink -  noop
+* BrokenLink -> SingleNode or MultiNode - no direct transition, the Config DB has to go through NotConfigured first.
+  * However, a vDVS service may first find out the state is BrokenLink, then admin does 2 commands and then vDVS service may directly transfer it's internal state to MultiNode or SingleNode
+
+
+### From SingleNode
+
+Transition from SingleNode means the admin live-reconfigured the Config DB.
+On the node where the operation happened, the admin CLI will refresh the vDVS service which will find out the new state from the DB.
+
+On other nodes, the discovery will find out about the change to `MultiNode` only, and will issue a warning to the admin  - since we do not know what was the intent.
+
+* SingleNode    -> Single Node      noop
+* SingleNode    -> NotConfigured    happened on `config -rm --local` only. Restart the service.
+* SingeNode     -> MultiNode        can happen on discovery. WARNING - Config changed. No action  we do not know what was the intent.
+* SingeNode     -> BrokenLink       ERROR - Throw exception
+
+### From MultiNode
+
+To transition from MultiNode, either the admin has to issue `config --rm` or there should be a FS or DB issue (i.e. the symlink was accidentally removed)
+
+* MultiNode      -> MultiNode       noop
+* MultiNode      -> NotConfigured   restart the service (on discovery or on CLI)
```

Comment:
(1) MultiNode->NotConfigured: `config rm --local --confirm`
(2)  MultiNode->SingleNode: `config rm --local --confirm`; `config init --local`
(3) `config --rm` does nothing, just explains how to rm shared DB :-). See the docs (.md)